src/containers/deletion-restorer.jsx

```diff
@@ -0,0 +1,67 @@
+import bindAll from 'lodash.bindall';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {connect} from 'react-redux';
+import {setRestore} from '../reducers/restore-deletion';
+
+/**
+ * DeletionRestorer component passes a restoreDeletion function to its child.
+ * It expects this child to be a function with the signature
+ *     function (restoreDeletion, props) {}
+ * The component can then be used to attach deletion restoring functionality
+ * to any other component:
+ *
+ * <DeletionRestorer>{(restoreDeletion, props) => (
+ *     <MyCoolComponent
+ *         onClick={restoreDeletion}
+ *         {...props}
+ *     />
+ * )}</DeletionRestorer>
+ */
+class DeletionRestorer extends React.Component {
+    constructor (props) {
+        super(props);
+        bindAll(this, [
+            'restoreDeletion'
+        ]);
+    }
+    restoreDeletion () {
+        if (typeof this.props.restore === 'function') {
+            this.props.restore();
+            this.props.dispatchUpdateRestore({restoreFun: null, deletedItem: ''});
+        }
+    }
+    render () {
+        const {
+            /* eslint-disable no-unused-vars */
+            children,
+            /* eslint-enable no-unused-vars */
+            ...props
+        } = this.props;
+        props.restorable = typeof this.props.restore === 'function';
+        props.deletedItem = this.props.deletedItem;
```

Comment:
I think in this case it's ok because `props` is a new object, but assigning to props doesn't look right, since most of the time it should be treated as immutable. Also, isn't `props.deletedItem = this.props.deletedItem` a noop?

I think a more consistent pattern would be something like:

```jsx
return this.props.children(this.restoreDeletion, {
    ...props,
    restorable: typeof this.props.restore === 'function
});
```